### PR TITLE
Load reviews file added

### PR DESCRIPTION
**Pull request by angelikavilde**
- Updated the schema for reviews to include 2 new columns
- Removed one column from extract and transform
- Added try and exceptions to the reviews pipeline with connection errors
- Changed time format for column in data to date format from timestamp

### CODE REVIEW

File: pipeline_reviews/extract.py [hunk 1]
```diff
@@ -44,43 +42,39 @@ def get_reviews_for_game(game_id: int, cursor: str) -> dict:
...
+
+                if not "error" in list(api_response.keys()):
+                    cursor = api_response["next_cursor"]
+                    page_reviews = api_response["reviews"]
+                    if not page_reviews or cursor in cursor_list:
+                        break
+                    if not cursor in cursor_list:
+                        cursor_list.append(cursor)
+                    all_reviews.extend(page_reviews)
+    make_csv_files(all_reviews)
```
Comment (angelikavilde) [inline pipeline_reviews/extract.py]: Just to note - this was changed mainly due to taking out the continue statements as they're classed as unprofessional. Also the removal of some data affected it but not the structure of the function.

**Review by DMcCallion — APPROVED** [review]
Looks very clear - nice work!

File: pipeline_reviews/load.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    cache_dict = dict()
+    reviews_df["game_id"] = reviews_df["game_id"].apply(
+        lambda row: get_game_ids(conn, row, cache_dict))
+    reviews_df = remove_empty_rows(reviews_df)
+    return reviews_df
+
+
+def get_game_ids(conn: connection, app_id: int, cache: dict) -> int | None:
+    """Returns game_id from game table from db (foreign key)"""
+    if str(app_id) in cache.keys():
```
Comment (gawsalya) [inline pipeline_reviews/load.py:27]: nice use of cache!

File: pipeline_reviews/load.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        return cache[str(app_id)]
+    try:
+        with conn.cursor() as cur:
+            cur.execute("""SELECT game_id FROM game WHERE app_id = %s""", app_id)
+            game_id = cur.fetchone()
+            conn.close()
+            game_id = game_id[["game_id"]]
+            cache[str(app_id)] = game_id
```
Comment (gawsalya) [inline pipeline_reviews/load.py]: close cursor here

File: pipeline_reviews/load.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    except (Error, TypeError) as e:
+        print("Error at load: ", e)
+        return None
+    return game_id
+
+
+def move_reviews_to_db(conn: connection, reviews_df: DataFrame) -> None:
+    """Moves all reviews into the database"""
+    data_to_insert = [tuple(row[1:]) for row in reviews_df.values]
+    execute_batch(conn.cursor(), """INSERT INTO review (game_id, review_text, review_score, review_date,
```
Comment (gawsalya) [inline pipeline_reviews/load.py]: not sure if you need to close cursor here as you've used it different but would double check

**Comment by gawsalya** [comment]
just need to close the cursor -- that's all!

File: pipeline_reviews/load.py (hunk 3 above)
Comment (angelikavilde) [reply pipeline_reviews/load.py]: So, connection needs closing as it was not entered and closed but cursor here was. So this is not needed here due to the "with" statement. If you ever mocked it, you could have noticed that you need to do `.__enter__` well this is for cursor to be opened but it also does `.__exit__ `(this you might have not seen with mocking but if you ever created mocking classes for it, you'd have to make one of this too as it does close the cursor). It does however not close the connection :) but this is added

File: pipeline_reviews/load.py (hunk 4 above)
Comment (angelikavilde) [reply pipeline_reviews/load.py]: After looking at the source code for it, it does seem like it should be closed here yes. Good catch!

**Pull request merged**